homeassistant/components/simplisafe/__init__.py

```diff
@@ -240,3 +244,73 @@ def __init__(self, hass, config_entry, systems):
         tasks = [self._update_system(system) for system in self.systems.values()]
 
         await asyncio.gather(*tasks)
+
+
+class SimpliSafeEntity(Entity):
+    """Define a base SimpliSafe entity."""
+
+    def __init__(self, system, name, *, serial=None):
+        """Initialize."""
+        self._async_unsub_dispatcher_connect = None
+        self._attrs = {ATTR_SYSTEM_ID: system.system_id}
+        self._name = name
+        self._online = True
+        self._system = system
+
+        if serial:
+            self._serial = serial
+        else:
+            self._serial = system.serial
+
+    @property
+    def available(self):
+        """Return whether the entity is available."""
+        # We can easily detect if the V3 system is offline, but no simple check exists
+        # for the V2 system. Therefore, we mark the entity as available if:
+        #   1. We can verify that the system is online (assuming True if we can't)
+        #   2. We can verify that the entity is online
+        system_offline = self._system.version == 3 and self._system.offline
+        return not system_offline and self._online
+
+    @property
+    def device_info(self):
+        """Return device registry information for this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._system.system_id)},
+            "manufacturer": "SimpliSafe",
+            "model": self._system.version,
+            "name": self._name,
+            "via_device": (DOMAIN, self._system.serial),
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attrs
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return f"{self._system.address} {self._name}"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return self._serial
```

Comment:
Does the unique_id change or stay the same?

Reply:
See the breaking change notes above – it changes.